### PR TITLE
Update Makefile targets for running gdb

### PR DESCRIPTION
**Pull request by cervantesyu**
- b2g process can be killed and restarted in gdb using run-gdb target
- no need to remount and rename b2g binary
- no need to generate .gdbinit file
- fix getting pid on galaxy s2

### CODE REVIEW

**Comment by andreasgal** [comment]
How do I run or restart the executable in gdb? Can you please create a documentation page for this on the wiki?

**Comment by cervantesyu** [comment]
after make run-gdb, the process is not yet started. Just 'run' in gdb as debugging a local .application. You may 'kill' or 'restart' the process in gdb as well.

**Comment by andreasgal** [comment]
Great. Explain in the wiki how to debug with a few simple examples and then I merge this bug. Thanks!

**Comment by cervantesyu** [comment]
Wiki page created: https://wiki.mozilla.org/B2G/gdb

File: Makefile [hunk 1]
```diff
@@ -544,51 +544,58 @@ GDBINIT=/tmp/gdbinit
 GDB=$(abspath glue/gonk/prebuilt/linux-x86/tegra-gdb/arm-eabi-gdb)
 B2G_BIN=/system/b2g/b2g
 
+ifeq "$(GONK)" "generic"
+PID_IDX=2
+endif
+
+ifeq "$(GONK)" "galaxys2"
+PID_IDX=1
+endif
```
Comment (ThinkerYzu) [inline Makefile]: Yesterday, @shianyow showed me a command "pidof" that is new for me.  You can switch to "pidof" instead of hacking here.

**Comment by mvines** [comment]
"pidof" isn't available on some versions of toolbox it seems (Akami and Maguro for example.)   I install busybox on device right away anyway so this doesn't bug me much but others might be getting tripped up here

**Comment by cervantesyu** [comment]
if toolbox will be available on all platforms, then it is easier to handle the inconsistence by getting ps output from toolbox:
toolbox ps and pid will always be at position 2.

**Comment by andreasgal** [comment]
We don't have the toolbox for now so lets just stick to the hack above

**Comment by shianyow** [comment]
@CervantesYu thanks, I've merged to latest Makefile.

**Pull request closed without merging**